src/Markdig/Helpers/ThrowHelper.cs

```diff
@@ -10,45 +10,46 @@ namespace Markdig.Helpers
     [ExcludeFromCodeCoverage]
     internal static class ThrowHelper
     {
+        public static void ArgumentNullException(string paramName) => throw new ArgumentNullException(paramName);

```

Comment:
I might be missing something, but shouldn't all these methods be marked as `[MethodImpl(MethodImplOptions.NoInlining)]`?

Reply:
They could be, but the Jit will never find inlining methods that always throw profitable afaik

Reply:
Ah I see, you're relying more on the JIT to do that for you. I guess I might just be overly cautious to always add that manually, just in case the code ends up running on some runtime with an older or less optimized JIT. But what you said makes perfect sense, yeah! 👍

Reply:
I can add them to document that that is the intent

Reply:
Yeah, it should be handled automatically in CoreCLR now (see https://github.com/dotnet/coreclr/pull/6103), but I guess it wouldn't hurt considering the lib targets .NET Standard, so it could be running on everything from .NET Framework to other runtimes as well 😊